### ISSUE TITLE
Array alignment

### ISSUE THREAD

**Issue by daurnimator** [ziglang/zig#2673]
As part of my module's public members, I want to expose a type that is an array of `u8` with `align(4)`.
This currently fails with 
```
error: align qualifier invalid on array type
pub const State = [48]align(4)u8;
```

**Comment by andrewrk** [comment]
The syntax is 

`pub const state align(4) = [48]u8{...}`

Or

`pub const state: [48]u8 align(4) = undefined`

The alignment is a property of the variable (memory location), not the type.

**Comment by daurnimator** [comment]
e.g. I have a function:
```zig
pub const State = [48]u8;
pub fn foo(state: *align(4)State) void {
```

And I want a user to be able to write:
```zig
fn bar() {
    var state: State = StateInitializer;
    foo(&state);
}
```

Then they get an alignment error.

How can they fix it?

**Comment by andrewrk** [comment]
They would have to put the alignment on the `state` variable like this:

```zig
    var state: State align(4) = StateInitializer;
```

If you want to encapsulate this into a type you'll need to use a struct and this issue needs to be implemented: #1512

**Comment by daurnimator** [comment]
> If you want to encapsulate this into a type you'll need to use a struct and this issue needs to be implemented: #1512

Thanks. I'm working around it for now by lying about the type (saying it's a `const State = [48 / 4]u32;` instead) and using `@sliceToBytes` everywhere.
